### PR TITLE
Switch to non sequential public ids

### PR DESCRIPTION
**Pull request by ctwoolsey**

### CODE REVIEW

File: src/main/java/svc/gson/configuration/WebMvcConfig.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+			 JsonArray jA = new JsonParser().parse(jsonArrayString).getAsJsonArray();
+			 jsonObj.add(fieldToAdd, jA);
+		 }
+		 
+		 public JsonObject getJsonObj(){
+			 return jsonObj;
+		 }
+	 }
+	 
+	 private class CourtSerializer implements JsonSerializer<Court> {
```
Comment (abobwhite) [inline src/main/java/svc/gson/configuration/WebMvcConfig.java]: It'd be easier to follow if you put all these serializers into their own files in a new serializer package or something.

File: src/main/java/svc/hashids/HashidsConfig.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package svc.hashids;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class HashidsConfig {
+	@Value("${stlcourts.security.municipalitySalt}")
```
Comment (abobwhite) [inline src/main/java/svc/hashids/HashidsConfig.java]: Can you simplify this into 1 bean that provides a new object of your creation (maybe "Hasher" or something). That class would have an encrypt/decrypt (or hash/unhash) function that would take 1) the class type and 2) the id. Then in that function, you can use a switch to determine which hashids object to use (which you would want to statically create in this "Hasher". It buries the logic and gives you a clean point of entry to all areas that need this functionality

File: src/main/java/svc/models/Municipality.java [hunk 3]
```diff
@@ -3,8 +3,10 @@
 
 import java.util.List;
 
+import svc.types.HashableEntity;
+
 public class Municipality {
-	public Long id;
+	public HashableEntity<Municipality> id;
 	public String name;
 	public List<Long> courts;
```
Comment (abobwhite) [inline src/main/java/svc/models/Municipality.java]: These need to be HashableEntity objects too

File: src/main/java/svc/security/HashUtil.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	@Value("${stlcourts.security.judgeSalt}")
+	String judgeSalt;
+	@Value("${stlcourts.security.courtSalt}")
+	String courtSalt;
+	
+	public Hashids municipalityHashids;
+	public Hashids judgeHashids;
+	public Hashids courtHashids;
+	
+	private void initializeSalts(){
```
Comment (abobwhite) [inline src/main/java/svc/security/HashUtil.java]: Why is this function necessary? The 3 Hashids objects should be able to be created statically, no? We shouldn't have to new them up each time the function is called. Then encode/decode can be static. The whole util should be able to be static. You may to make the salts final or `@Shared` or something

File: src/main/java/svc/types/HashableEntity.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+	public Class<CLASS> type;
+	
+	public Long value;
+	
+	public HashableEntity(Class<CLASS> cls, Long id){
+		this.value = id;
+		this.type = cls;
+	}
+	
+	public Long getValue(){
```
Comment (abobwhite) [inline src/main/java/svc/types/HashableEntity.java]: You don't need BOTH public properties AND getters/setters. Just use one or the other

File: src/main/java/svc/security/HashUtil.java (hunk 4 above)
Comment (ctwoolsey) [inline src/main/java/svc/security/HashUtil.java]: So I found that @Value is only read at runtime off of an @Inject.  If I try to use the salt values outside of a method, they return null.  Same goes if I create a constructor they are still null there.  So only during runtime when a HashUtil.encode is called are the @Value salts valid.

Comment (abobwhite) [inline src/main/java/svc/security/HashUtil.java]: OK - let's re-evaluate later. It's pretty inefficient to build these things new over and over with the same salt

File: src/main/java/svc/security/HashUtil.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+			municipalityHashids = new Hashids(municipalitySalt,8);
+			judgeHashids = new Hashids(judgeSalt,8);
+			courtHashids = new Hashids(courtSalt,8);
+		}
+	}
+	
+	public String encode(Class<?> fieldClass, long value){
+		initializeSalts();
+		String hashedValue = "";
+		switch (fieldClass.getTypeName()){
```
Comment (abobwhite) [inline src/main/java/svc/security/HashUtil.java]: Use getSimpleName instead - that way this code doesn't break if we rename a package

https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getSimpleName--

File: src/main/java/svc/security/HashUtil.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+			case "svc.models.Court":
+				hashedValue = courtHashids.encode(value);
+				break;
+		}
+		return hashedValue;
+	}
+	
+	public long decode(Class<?> fieldClass, String hashValue){
+		long idValue = 0L;
+		switch (fieldClass.getTypeName()){
```
Comment (abobwhite) [inline src/main/java/svc/security/HashUtil.java]: Same here

File: src/main/java/svc/security/HashUtil.java (hunk 4 above)
Comment (ctwoolsey) [inline src/main/java/svc/security/HashUtil.java]: I agree.  I just couldn't figure out how to do it differently.

**Pull request merged**